#**Live Code 3**

# Introduction

Nama  : Fitri Octaviani

Batch  : HCK 006

# **Problem Statement**

> Kejahatan yang terjadi di kota Austin Smart City pada tahun 2016

# Success Criteria

> S : Mengurangi kasus tingkat kejahatan yang belum terselesaikan pada tahun 2016.
>
> M : Ketika berhasil mengurangi jumlah kasus kejahatan yang terselesaikan di kota Austin pada tahun 2016.
>
> A : Ketika berhasil mengurangi kasus kejahatan di kota Austin pada tahun 2016.
>
> R : Kepolisian mampu mengurangi kejahatan sedikit demi sedikit dari tiap bulannya.
>
> T : Target yang dicapai adalah 2017.


# Data Loading

In [1]:
from google.colab import auth
from google.cloud import bigquery
auth.authenticate_user()
print('Authenticated')

project_id = "grad1-388605" #GUNAKAN GCP PROJECT-ID KALIAN MASING-MASING
client = bigquery.Client(project=project_id)

Authenticated


## SOAL 1

> Tampilkan tabel crime sampai 5 baris pertama

In [2]:
# Menampilkan tabel crime sampai 5 baris pertama

tabel_crime = client.query('''
SELECT *
FROM `bigquery-public-data.austin_crime.crime`
LIMIT 5

''').to_dataframe()
tabel_crime

unique_key     address  census_tract            clearance_date  \
0   2015150483  Austin, TX           NaN 2015-01-27 12:00:00+00:00   
1  20151542027  Austin, TX           NaN 2015-07-24 12:00:00+00:00   
2  20152001758  Austin, TX           NaN 2015-08-15 12:00:00+00:00   
3  20151340282  Austin, TX           NaN 2015-06-15 12:00:00+00:00   
4  20151871268  Austin, TX           NaN 2015-10-27 12:00:00+00:00   

       clearance_status  council_district_code  \
0           Not cleared                   <NA>   
1           Not cleared                   <NA>   
2           Not cleared                   <NA>   
3  Cleared by Exception                   <NA>   
4           Not cleared                   <NA>   

                      description district  latitude  longitude location  \
0  RAPE                                  B       NaN        NaN     None   
1  RAPE                                  D       NaN        NaN     None   
2  RAPE                                  F       NaN        NaN     None   
3  RAPE                                 UK       NaN        NaN     None   
4  RAPE                                 UK       NaN        NaN     None   

  location_description primary_type                 timestamp  x_coordinate  \
0                  nan         Rape 2015-01-15 12:00:00+00:00          <NA>   
1                  nan         Rape 2015-06-03 12:00:00+00:00          <NA>   
2                  nan         Rape 2015-07-19 12:00:00+00:00          <NA>   
3                  nan         Rape 2015-05-14 12:00:00+00:00          <NA>   
4                  nan         Rape 2015-07-06 12:00:00+00:00          <NA>   

   y_coordinate  year zipcode  
0          <NA>  2015    None  
1          <NA>  2015    None  
2          <NA>  2015    None  
3          <NA>  2015    None  
4          <NA>  2015    None

Dalam tabel crime tersebut kolom unique_key, address,	census_tract,	clearance_date,	clearance_status,	council_district_code,	description,	district,	latitude,	longitude,	location,	location_description,	primary_type,	timestamp,	x_coordinate,	y_coordinate,	year,	zipcode.

In [3]:
import pandas as pd
import numpy as np

In [4]:
df1 = pd.read_csv ('/content/sample_data/P0LC3_Fitri Octaviani_1.csv')
df1

unique_key     address  census_tract                  clearance_date  \
0   2015150483  Austin, TX           NaN  2015-01-27 12:00:00.000000 UTC   
1  20151542027  Austin, TX           NaN  2015-07-24 12:00:00.000000 UTC   
2  20152001758  Austin, TX           NaN  2015-08-15 12:00:00.000000 UTC   
3  20151340282  Austin, TX           NaN  2015-06-15 12:00:00.000000 UTC   
4  20151871268  Austin, TX           NaN  2015-10-27 12:00:00.000000 UTC   

       clearance_status  council_district_code  \
0           Not cleared                    NaN   
1           Not cleared                    NaN   
2           Not cleared                    NaN   
3  Cleared by Exception                    NaN   
4           Not cleared                    NaN   

                      description district  latitude  longitude  location  \
0  RAPE                                  B       NaN        NaN       NaN   
1  RAPE                                  D       NaN        NaN       NaN   
2  RAPE                                  F       NaN        NaN       NaN   
3  RAPE                                 UK       NaN        NaN       NaN   
4  RAPE                                 UK       NaN        NaN       NaN   

   location_description primary_type                       timestamp  \
0                   NaN         Rape  2015-01-15 12:00:00.000000 UTC   
1                   NaN         Rape  2015-06-03 12:00:00.000000 UTC   
2                   NaN         Rape  2015-07-19 12:00:00.000000 UTC   
3                   NaN         Rape  2015-05-14 12:00:00.000000 UTC   
4                   NaN         Rape  2015-07-06 12:00:00.000000 UTC   

   x_coordinate  y_coordinate  year  zipcode  
0           NaN           NaN  2015      NaN  
1           NaN           NaN  2015      NaN  
2           NaN           NaN  2015      NaN  
3           NaN           NaN  2015      NaN  
4           NaN           NaN  2015      NaN

## SOAL 2

> Apa saja jenis tindak kejahatan yang terjadi di kota Austin selama tahun 2016?

In [5]:
Jenis_Kejahatan = client.query('''
SELECT DISTINCT primary_type as jenis_kejahatan
FROM bigquery-public-data.austin_crime.crime
where EXTRACT(YEAR FROM timestamp) = 2016

''').to_dataframe()
Jenis_Kejahatan

jenis_kejahatan
0            Rape
1           Theft
2         Robbery
3     Agg Assault
4        Burglary
5      Auto Theft
6          Murder

Jenis tindak kejahatan yang terjadi di kota Asutin ada Rape, Theft, Robbery, Agg Assault, Burglary, Auto Theft, dan Murder. Dengan berbagai kejahatan ini maka kepolisian lebih teliti lagi perihal keamanan di kota Austin.

## SOAL 3

> Berapa banyak kasus tindak kejahatan berdasarkan kasus pembersihannya (clearance status) selama tahun 2016?

In [6]:
kasus_pembersihan = client.query('''
SELECT clearance_status, COUNT(*) AS jumlah_clear
FROM bigquery-public-data.austin_crime.crime
where EXTRACT(YEAR FROM timestamp) = 2016
GROUP BY clearance_status

''').to_dataframe()
kasus_pembersihan

clearance_status  jumlah_clear
0           Not cleared         29715
1  Cleared by Exception          1014
2                  None          1719
3     Cleared by Arrest          5013

Pada tahun 2016 ini tindak kejahatan yang paling banyak adalah dengan status not cleared dengan jumlah 29715. Hal ini polisi sangat diperlukan untuk lebih menyelesaikan kasus yang belum terselesaikan.

## SOAL 4

> Jenis kejahatan apa saja yang paling banyak belum terselesaikan kasusnya di tahun 2016?

In [7]:
belum_terselesaikan = client.query('''
SELECT primary_type as Jenis_kejahatan, clearance_status, COUNT (unique_key) as banyak_kejahatan
FROM bigquery-public-data.austin_crime.crime
where clearance_status = 'Not cleared'
      AND EXTRACT(YEAR FROM timestamp) = 2016
GROUP BY primary_type, clearance_status
ORDER BY banyak_kejahatan desc

''').to_dataframe()
belum_terselesaikan

Jenis_kejahatan clearance_status  banyak_kejahatan
0           Theft      Not cleared             21894
1        Burglary      Not cleared              4440
2      Auto Theft      Not cleared              1540
3     Agg Assault      Not cleared               919
4         Robbery      Not cleared               560
5            Rape      Not cleared               361
6          Murder      Not cleared                 1

Jenis kejahatan yang paling banyak tidak dibersihkan adalah theft. Sehingga polisi perlu memerhatikan keamanan sekitar dan warga di kota austin pun harus lebih waspada.

## SOAL 5

Di bulan apa di tahun 2016 kasus pencurian (theft) sangat banyak terjadi?

In [8]:
kasus_pencurian = client.query('''
SELECT primary_type, EXTRACT (month from (clearance_date)) as Bulan,
COUNT (*) as kasus
FROM bigquery-public-data.austin_crime.crime
WHERE
EXTRACT(YEAR FROM clearance_date) = 2016
AND primary_type = 'Theft'
GROUP BY primary_type, Bulan
ORDER BY kasus desc

''').to_dataframe()
kasus_pencurian

primary_type  Bulan  kasus
0         Theft      8   2375
1         Theft     11   2345
2         Theft      5   2185
3         Theft      1   2161
4         Theft     12   2091
5         Theft      3   2060
6         Theft      6   2038
7         Theft      4   2028
8         Theft     10   2020
9         Theft      2   1986
10        Theft      9   1957
11        Theft      7   1824

Kasus theft paling banyak terjadi adalah pada bulan ke-8 dengan total 2375.

## SOAL 6


Di distrik apa yang paling banyak terjadi kasus pencurian selama 2016?

In [9]:
distrik = client.query('''

SELECT district,
        Count (*) as theft
FROM bigquery-public-data.austin_crime.crime
WHERE
      EXTRACT(YEAR FROM clearance_date) = 2016
      and primary_type = 'Theft'
GROUP BY district
ORDER BY theft desc

''').to_dataframe()
distrik

district  theft
0         D   3839
1         B   3422
2         A   2929
3         I   2898
4         F   2857
5         E   2836
6         H   2412
7         G   2111
8         C   1609
9        AP    119
10       UK     35
11       88      3

Dalam kota asutin, district yang paling banyak terjadi kasus pencurian adalah district D. Untuk mengurangi tingkat kejahatan ini maka perlu waspada.

# Poin Analisis

> Dalam kota austin memang banyak kejahatan yang terjadi, akan tetapi dalam kasus kejahatan itu paling banyak adalah theft dengan total kasus yang tidak terselesaikan senbanyak 21894. Hal ini perlu penegakan hukum yang ketat dan pencegahan tindak kejahatan yang baik agar dapat mengurangi tindakan kriminal yang terjadi. Selain itu polisi juga perlu memperhatikan keadaan sekitar agar tidak terjadinya kejahatan kriminal.